### PR TITLE
README: Flesh out and add a simple description

### PR DESCRIPTION
**Pull request by richlowe**

### CODE REVIEW

**Review by danmcd — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 
-This repository is a preliminary holding area for a new shrink-to-fit process
-for collaboration on larger scale illumos projects.  This README itself will
-get fleshed out soon, but in the meantime there is a table of initial IPD
-(illumos project discussion) documents below:
+IPDs are a shrink-to-fit description of larger scale (in either code size or
+impact) project wishing to make changes to illumos.  They should describe the
+what and the why and give people the opportunity to comment on both.
+
+An IPD is at its heart, just a README.md in a numbered directory in this
```
Comment (rzezeski) [inline README.md:7]: I guess I'm the only one who did an .adoc, haha. I'd say we should allow either. And if that's the case maybe rephrase this like "An IPD consists of either a markdown or asciidoc README file in a numbered directory".

File: README.md [hunk 2]
```diff
@@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
...
-[@jclulow](https://github.com/jclulow) or open an issue against this
-repository.
+Contributions are welcome.  A good rule of thumb as to whether you _should_
+have an IPD is whether you are making a change with high impact to other
+developers or users (introducing or removing a supported platform, doing
+something with non-obvious compatibility constraints), or engaging in a
+long-term project that will likely integrate in pieces, to provide the overall
+picture.
+
+For your first contribution, you might want to just submit a pull request to
```
Comment (rzezeski) [inline README.md:60]:
```suggestion
For your first contribution, you should submit a pull request to
```

File: README.md [hunk 3]
```diff
@@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
 
 ## Contributing
 
-Contributions are welcome!  In the short term, you can reach out to
-[@jclulow](https://github.com/jclulow) or open an issue against this
-repository.
+Contributions are welcome.  A good rule of thumb as to whether you _should_
+have an IPD is whether you are making a change with high impact to other
```
Comment (rzezeski) [inline README.md:54]:
```suggestion
write an IPD is whether you are making a change with high impact to other
```

File: README.md [hunk 4]
```diff
@@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
...
+developers or users (introducing or removing a supported platform, doing
+something with non-obvious compatibility constraints), or engaging in a
+long-term project that will likely integrate in pieces, to provide the overall
+picture.
+
+For your first contribution, you might want to just submit a pull request to
+this repository.  Going forward if this is a thing that you will do again,
+we'll probably give you write access to this repository so you can just add
```
Comment (rzezeski) [inline README.md:62]:
```suggestion
we'll probably give you write access to this repository so you can add
```

File: README.md [hunk 5]
```diff
@@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
...
+
+An IPD has a short header block indicating authorship (that's you),
+sponsorship (we'll get to that), and state.
+
+### States
+
+#### predraft
+
+You've started writing your IPD and you want to share it narrowly, or even
+just to reserve your a number in this repository.  You're _predraft_, maybe
```
Comment (rzezeski) [inline README.md:75]:
```suggestion
just to reserve a number in this repository.  You're _predraft_, maybe
```

File: README.md [hunk 6]
```diff
@@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
...
+One or more people from the [illumos core
+team](https://illumos.org/docs/about/leadership/) have agreed that what you've
+described is a good thing, and that we should do it.  Your IPD is done and
+published (though is not immutable! If you find more information would be
+useful later, please add it!)
+
+### Sponsorship
+
+"Sponsor" is a weird word here, it's just the person or people on the illumos
+core team who were ok with your IPD.  Don't worry about it.
```
Comment (rzezeski) [inline README.md:95]: Rich, first I want to be clear I'm happy someone took the time to flesh out the IPD process a bit. But I can't help but read this and wonder why we even have sponsorship if it's something no one should worry about and it's "weird"? If we can't explain what it is to the general public, maybe we don't need it?

It seems like the sponsor is the illumos core member who agreed the IPD should be moved to the published state? If so that seems easy enough to explain.

**Review by rzezeski — APPROVED** [review]
Rich, thank you for taking the time to give some details on the IPD process. Overall I find this to be a major improvement to what we currently have, so feel free to ignore any and all comments I've made in the spirit of getting something better out there.

**Comment by danmcd** [comment]
Did this get added already?  If so should we close this PR?

**Comment by danmcd** [comment]
Looks like https://github.com/illumos/ipd/commit/d07373867c0594314b6f7e1b859b3b318892fcd7 sure is a copy of this...

**Comment by danmcd** [comment]
Yes.  This.patch and commit.patch have nothing really different.  Closing:

```
zathras(~)[2]% diff /tmp/pr.patch /tmp/commit.patch 
1c1
< From e5339efd0bc762501cdf825f0cf36fb42add8775 Mon Sep 17 00:00:00 2001
---
> From d07373867c0594314b6f7e1b859b3b318892fcd7 Mon Sep 17 00:00:00 2001
11c11
< index 66148cf..0089a94 100644
---
> index feeb134..b6fcc9f 100644
31c31
< @@ -47,6 +50,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
---
> @@ -48,6 +51,46 @@ get fleshed out soon, but in the meantime there is a table of initial IPD
zathras(~)[1]% 
```
